src/gordon_gcp/clients/http.py

```diff
@@ -83,9 +78,11 @@ def __init__(self, auth_client=None, session=None):
         if not is_valid:
             await self._auth_client.refresh_token()
 
-    async def request(self, method, url, params=None, body=None,
-                      headers=None, **kwargs):
-        """Make an asynchronous HTTP request.
+    async def request(self, method, url, params=None, body=None, headers=None,
+                      retry_type="none", tries=1, timeout=None,
+                      retry_predicate=None, predicate_args=None,
+                      predicate_kwargs=None, extra_401_retry=True):
```

Comment:
I think I may understand the motivation for this overall approach - multiple keyword arguments for the ability to address many use cases/configurability. Do I understand correctly?

I fear this might be too much at once, and I'd much prefer to see the most common case addressed first, then we add what we need when we need it.

May I ask: is there a reason you chose to not use decorators? I suspect maybe because decorators are evaluated at runtime, making it more difficult to configure. I think it might be possible ([docs](https://github.com/litl/backoff#runtime-configuration)) since plugins are loaded and config [should be available](https://github.com/spotify/gordon/blob/develop/gordon/plugins_loader.py#L67-L91). Perhaps you already tried that though.

Reply:
You're correct about the motivation.  I agree it might be too much, but it seemed like a small amount of code to add to gain more functionality.  Maybe the calculus changes with your suggestion below of using predicate_kwargs?

The decorator thing I don't think can work unless we move _request() into request() as an inner function, but I'm open to discussing it.  Certainly easier to read in that case.